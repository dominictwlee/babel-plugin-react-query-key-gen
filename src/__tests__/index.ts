import pluginTester from 'babel-plugin-tester';
import path from 'path';
import reactQueryStringKeyGenPlugin from '../index';

console.log(__dirname);

pluginTester({
  plugin: reactQueryStringKeyGenPlugin,
  pluginName: 'babel-plugin-react-query-string-key-gen',
  fixtures: path.join(__dirname, '__fixtures__'),
  tests: {
    'adds string key to the start of array queryKey': {
      code: `
        const { data } = useQuery([{id: 2}], () => getTodos());
      `,
      snapshot: true,
    },
    'Infer queryFn name from async/await function block': {
      code: `
        const { data: plainData } = useQuery(
        [{param1: 'abc'}],
        async () => {
          const blah = 'hello';
          const someOtherStuff = calculate();
          const result = await batchGetProductSwapList(productNumbers, idToken);
          return result;
        },
        { enabled: true }
      );
      `,
      snapshot: true,
    },
    'ignores function calls that are not useQuery': {
      code: `
        const { data } = useSomethingElse([{id: 2}], () => getTodos());
        const { blah } = useQueryObject({queryFn: 'hello', queryKey: 'helloWorld', options: {}});
        callMe(1, 2, 3);
      `,
      snapshot: true,
    },
  },
});
