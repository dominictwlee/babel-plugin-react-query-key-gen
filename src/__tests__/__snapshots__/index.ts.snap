// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-query-string-key-gen Infer queryFn name from async/await function block: Infer queryFn name from async/await function block 1`] = `

const { data: plainData } = useQuery(
  [{param1: 'abc'}],
  async () => {
    const blah = 'hello';
    const someOtherStuff = calculate();
    const result = await batchGetProductSwapList(productNumbers, idToken);
    return result;
  },
  { enabled: true }
);

      ↓ ↓ ↓ ↓ ↓ ↓

const { data: plainData } = useQuery(
  [
    'batchGetProductSwapList',
    {
      param1: 'abc',
    },
  ],
  async () => {
    const blah = 'hello';
    const someOtherStuff = calculate();
    const result = await batchGetProductSwapList(productNumbers, idToken);
    return result;
  },
  {
    enabled: true,
  }
);


`;

exports[`babel-plugin-react-query-string-key-gen adds string key to the start of array queryKey: adds string key to the start of array queryKey 1`] = `

const { data } = useQuery([{id: 2}], () => getTodos());

      ↓ ↓ ↓ ↓ ↓ ↓

const { data } = useQuery(
  [
    'getTodos',
    {
      id: 2,
    },
  ],
  () => getTodos()
);


`;

exports[`babel-plugin-react-query-string-key-gen ignores function calls that are not useQuery: ignores function calls that are not useQuery 1`] = `

const { data } = useSomethingElse([{id: 2}], () => getTodos());
const { blah } = useQueryObject({queryFn: 'hello', queryKey: 'helloWorld', options: {}});
callMe(1, 2, 3);

      ↓ ↓ ↓ ↓ ↓ ↓

const { data } = useSomethingElse(
  [
    {
      id: 2,
    },
  ],
  () => getTodos()
);
const { blah } = useQueryObject({
  queryFn: 'hello',
  queryKey: 'helloWorld',
  options: {},
});
callMe(1, 2, 3);


`;
